function-server/main.py

```diff
@@ -47,10 +52,19 @@ def process_msg(msg, handle):
     try:
         sys.stderr = stderr
         sys.stdout = stdout
-        r = handle(msg['context'], msg['payload'])
+        with ThreadPoolExecutor(max_workers=1) as executor:
+            future = executor.submit(handle, msg['context'], msg['payload'])
+            timeout = msg['context'].get('timeout', None)
+            r = future.result(None if timeout == None or timeout == 0 else timeout / 1000)
```

Comment:
I just realized you could do: 
```
timeout = msg['context'].get('timeout', 0)
r = future.result(timeout / 1000)
```

Reply:
Unfortunately that doesn't work. It is actually setting a timeout in the case that the context does not include a timeout.

Reply:
Ah you're right, 0 doesn't work it must be `None`.